Sources/AppStoreConnectCLI/Commands/Reports/DownloadSalesAndTrendsReportsCommand.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import AppStoreConnect_Swift_SDK
+import ArgumentParser
+import FileSystem
+
+struct DownloadSalesAndTrendsReportsCommand: CommonParsableCommand {
+
+    typealias Filter = DownloadSalesAndTrendsReports.Filter
+
+    static var configuration = CommandConfiguration(
+        commandName: "sales",
+        abstract: "Download sales and trends reports filtered by your specified criteria.")
+
+    @OptionGroup()
+    var common: CommonOptions
+
+    @Argument(help: "Frequency of the report to download. (\(Filter.Frequency.allCases.description))")
+    var frequency: Filter.Frequency
+
+    @Argument(help:
+        ArgumentHelp(
+            "The report date to download.",
+            discussion: "The date is specified in the YYYY-MM-DD format for all report frequencies except DAILY, which does not require a specified date."
+        )
+    ) var reportDate: String
+
+    @Argument(help: "The report to download.  (\(Filter.ReportType.allCases.description))")
+    var reportType: Filter.ReportType
+
+    @Argument(help: "The report sub type to download.  (\(Filter.ReportSubType.allCases.description))")
+    var reportSubType: Filter.ReportSubType
+
+    @Argument(help: "Your vendor number.")
+    var vendorNumber: String
+
+    @Argument(help: "The downloaded report file name.")
+    var outputFilename: String
+
+    @Option(help: "The version of the report.")
+    var version: [String]
```

Comment:
Would it be possible to have a default value for version for each report type?  There are only certain combinations of `reportType`, `reportSubType` and `version` that are valid. 

See: https://developer.apple.com/documentation/appstoreconnectapi/download_sales_and_trends_reports

We should validate the input better.  Maybe we should actually just have our own set of enums that restrict the input more.  As it stands certain combinations will not work.

Maybe something more like this:
```
asc reports sales_daily 12345678 ./file
asc reports sales_weekly 12345678 ./file
asc reports newsstand_weekly 12345678 ./file
```